### PR TITLE
[SCSB-145] require Python 3.10

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-145](https://jira.stsci.edu/browse/SCSB-145)

propagate Astropy's deprecation of Python 3.9 to downstream packages

> [!NOTE]
> this PR was generated automatically by ``batchpr`` :robot:

### CODE REVIEW

**Comment by tapastro** [comment]
This should probably also update the README/Docs to change the listed minimum version allowed - the alternative is to change that once the next build is done, but that risks it being forgotten about. Best practices?

**Comment by zacharyburnett** [comment]
Sorry for the oversights! I'm using this ticket to test making automated PRs with `batchpr`, so hopefully the next Python version bump will be easier

File: pyproject.toml
Comment (emolter) [inline pyproject.toml:1]: Line 16 should be removed; Python 3.9 should no longer be one of the classifiers

**Review by emolter — COMMENTED** [review]
There are also a bunch of references to Python 3.9 in the install.rst file that should be removed

File: README.md [hunk 1]
```diff
@@ -9,16 +9,14 @@
 
 ![STScI Logo](docs/_static/stsci_logo.png)
 
-**JWST requires a C compiler for dependencies and is currently limited to Python 3.9, 3.10 or 3.11.**
+> [!IMPORTANT]
+> JWST requires a C compiler for dependencies and is currently limited to Python 3.10, 3.11, or 3.12.
 
-**Until Python 3.12 is supported, fresh conda environments will require setting the
```
Comment (hbushouse) [inline README.md:14]: Do we actually support py3.12 now (or will support it in the B10.2 release)?  i.e. does everything actually work correctly now under 3.12?

Comment (braingram) [reply README.md:14]: I'm not sure if anyone has run the regtests with python 3.12. @zacharyburnett would changing the following be enough to convince jenkins to use python 3.12?
https://github.com/spacetelescope/jwst/blob/e1b07728a2d610e2eae685f36ac9f18905b8e94b/JenkinsfileRT#L46
The unit tests have been running with python 3.12:
https://github.com/spacetelescope/jwst/actions/runs/8345372313/job/22840164182

We have one open PR that is python 3.12 related (which addresses a `DeprecationWarning` introduced in that version):
https://github.com/spacetelescope/jwst/pull/8343

Comment (zacharyburnett) [reply README.md:14]: yes, that should be enough; I can make a separate PR with that

**Comment by zacharyburnett** [comment]
> There are also a bunch of references to Python 3.9 in the install.rst file that should be removed

fixed in a2fdc772d7a57587423de3387519ddfc6fccd491

**Review by hbushouse — APPROVED** [review]

**Pull request merged**